### DISCUSSION TITLE
Distributed Hypervisor over high speed interconnect

### DISCUSSION THREAD

**Discussion by ohault** [cloud-hypervisor/cloud-hypervisor#6893] [Ideas]
Using high speed interconnect (PCIe interconnect, n x 10GbE, IB, ...), a major feature would be to support the concept of a Distributed Hypervisor.

There are existing projects like : 
- https://giantvm.github.io/
- https://github.com/systems-nuts/FragVisor

Linux Foundation should perhaps consider to fellowship a project to create a generic Distributed Virtual Machine technology as a very thin layer.

**Comment by iggy** [comment]
Something tells me the devs will see this as outside the scope of cloud-hypervisor. This concept has come up in virtualization numerous times over the years and it's never gone beyond some uni driven hack that never goes anywhere. I think the closest anybody got to something marginally useful was OpenSSI. Every other project has fizzled as soon as people started trying to solve the hard problems. I've never seen anybody actually show an appreciable speedup over something specific to a problem (mapreduce, mpi, etc).

**Reply by ohault** [reply]
> Something tells me the devs will see this as outside the scope of cloud-hypervisor. This concept has come up in virtualization numerous times over the years and it's never gone beyond some uni driven hack that never goes anywhere. I think the closest anybody got to something marginally useful was OpenSSI. Every other project has fizzled as soon as people started trying to solve the hard problems. I've never seen anybody actually show an appreciable speedup over something specific to a problem (mapreduce, mpi, etc).

@iggy, the main differences are that nowadays we have hypervisor technologies, high-speed interconnect, and CXL protocols.
